### ISSUE TITLE
Async std::os::unix::fs::FileExt read_at/write_at

### ISSUE THREAD

**Issue by Xorlev** [async-rs/async-std#576]
Hi folks,

I was curious whether async-std would be interested in an async port of `std::os::unix::fs::FileExt`. I'm writing a service right now that'd benefit from such a thing -- currently I've been doing it myself by using std::os::unix::fs::FileExt::read_at with `spawn_blocking`, but it'd be neat to be included in async-std.

If there's interest in this, I could try my hand at a PR.

**Comment by stjepang** [comment]
Yes! :) We haven't quite finished the `async_std::os` module. A PR with `FileExt` would be very welcome!

**Comment by vlovich** [comment]
I want to try my hand here. Organizationally, does this get added to src/os/windows/fs.rs and src/os/unix/fs.rs?
